### PR TITLE
Integrate tile map element pointers usage

### PR DESCRIPTION
**Pull request by janisozaur**

### CODE REVIEW

File: src/ride/track.c [hunk 1]
```diff
@@ -244,6 +244,7 @@ const rct_trackdefinition gTrackDefinitions_INCORRECT[] = {
 #define TRACK_MAX_SAVED_MAP_ELEMENTS 1500
 
 rct_map_element **gTrackSavedMapElements = (rct_map_element**)0x00F63674;
+static rct_map_element **gMapElementTilePointersBackup = NULL;
```
Comment (IntelOrca) [inline src/ride/track.c]: Does this need to be global if its only used in ```backup_map()```?

Comment (janisozaur) [inline src/ride/track.c]: isn't `static` enough? What else should it be?

Comment (IntelOrca) [inline src/ride/track.c]: The ```g``` prefix is for global application variables (non-static), ```_``` for static.

Comment (IntelOrca) [inline src/ride/track.c]:
```
backedUpMap = backup_map();
// do stuff...
restore_map(backedUpMap);
```

**Comment by janisozaur** [comment]
done

**Pull request merged**